tests/basic/uncontrolled_randomness.rs

```diff
@@ -0,0 +1,216 @@
+use shuttle::rand::{thread_rng, Rng};
+use shuttle::scheduler::{DfsScheduler, Scheduler};
+use shuttle::sync::{Arc, Mutex};
+use shuttle::{check_uncontrolled_randomness, thread};
+use test_log::test;
+
+fn check_uncontrolled_randomness_custom_scheduler_and_config<F, S>(f: F, scheduler: S)
+where
+    F: Fn() + Send + Sync + 'static,
+    S: Scheduler + 'static,
+{
+    use shuttle::scheduler::UncontrolledRandomnessCheckScheduler;
+    use shuttle::Config;
+
+    let config = Config::default();
+
+    let scheduler = UncontrolledRandomnessCheckScheduler::new(scheduler);
+
+    let runner = shuttle::Runner::new(scheduler, config);
+    runner.run(f);
+}
+
+#[test]
+fn randomly_acquire_lock_shuttle_rand() {
+    check_uncontrolled_randomness(
+        || {
+            const NUM_THREADS: u32 = 10;
+
+            let lock = Arc::new(Mutex::new(0u64));
+            let threads: Vec<_> = (0..NUM_THREADS)
+                .map(|_| {
+                    let my_lock = lock.clone();
+
+                    thread::spawn(move || {
+                        let x = thread_rng().gen::<u64>();
+
+                        // Fail every x threads
```

Comment:
This comment is confusing -- there's no "failure" here, we're just picking one of every 10 threads to increment the shared counter.

In any case, why not just make every thread lock the counter, and add `x % 10` to it?  Why bother with the `if` ?

Reply:
If we have every thread lock the counter, then we will not catch that there is uncontrolled randomness. We are catching discrepancies in the locking of the mutex, not in the value of `num`.